frontend/src/app/modules/matches/matches-computations.js

```diff
@@ -17,9 +17,10 @@ export const generateMatchRatingChanges = (userId, userMatches) => userMatches.m
     }
 })
 
-export const generateRatingHistoryGraphForUser = (userMatches, userId, initialRating) => userMatches
-  .reverse()
-  .map(
+export const generateRatingHistoryGraphForUser = (userMatches, userId, initialRating) => {
+  const reversedUserMatches = userMatches.reverse()
+  return [{ x: userMatches[0] ? new Date().setDate(userMatches[0].date.getDate() - 1) : new Date(), y: initialRating }]
```

Comment:
I'm assuming you want the initial rating to be on the graph one day before the first match, however I'm not sure the expression:
`new Date().setDate(userMatches[0].date.getDate() - 1)` 
would work in this case. You're just creating an object with current date (`new Date`) then setting the `date` which is actually just the day of the month (yaay javascript), so if the `userMatches[0].date` is say on `2018-6-6` the resulting initial date (today) would be `2019-9-5` and not `2018-6-5`.

Reply:
Thanks, you pointed me to some minor issues with the code that I'll fix, but the thing that you point out works as expected:
```
> const today = new Date();
undefined
> today;
2019-09-02T09:40:24.694Z
> const yesterdayTs = (new Date()).setDate(today.getDate() - 1);
undefined
> const yesterday = new Date(yesterdayTs);
undefined
> yesterday;
2019-09-01T09:40:28.670Z
> const lastDayOfAugustTs = (new Date()).setDate(yesterday.getDate() - 1);
undefined
> const lastDayOfAugust = new Date(lastDayOfAugustTs);
undefined
> lastDayOfAugust;
2019-08-31T09:41:05.698Z
```

Reply:
You're right actually, the method works, except I am using it wrong... :D

Reply:
No worries man, happens to the best of us